opentelemetry/proto/common/v1/common.proto

```diff
@@ -60,3 +60,50 @@ message InstrumentationLibrary {
   string name = 1;
   string version = 2;
 }
+
+// ValueType is the enumeration of possible types that value can have.
+// TODO: consolidate this with AttributeKeyValue.ValueType.
+enum ValueType {
+  STRING  = 0;
+  INT     = 1;
+  DOUBLE  = 2;
+  BOOL    = 3;
+  LIST    = 4;
+  MAP     = 5;
+};
+
+// AnyValue is a value that is used to store values that can be one of the supported
+// value types.
+message AnyValue {
+  // type of the value.
+  ValueType type = 1;
+
+  // Only one of the following fields is supposed to contain data (determined by `type` field).
+  // This is deliberately not using Protobuf `oneof` for performance reasons (verified by benchmarks).
+
+  bool bool_value = 2;
```

Comment:
nit: I wonder why the order of fields doesn't match the order in enum =).

Reply:
It reduces memory usage because of better packing of fields due to struct field alignment rules.

Reply:
good to know. Worth mentioning in comments for future reference.